crates/ra_prof/src/lib.rs

```diff
@@ -255,6 +255,39 @@ impl Drop for Scope {
     }
 }
 
+/// A wrapper around https://github.com/AtheMathmo/cpuprofiler
+///
+/// It can be used to capture sampling profiles of sections of code.
+/// It is not exactly out-of-the-box, as it relies on gperftools.
+/// See the docs for the crate for more!
+#[derive(Debug)]
+pub struct CpuProfiler {
+    _private: (),
+}
+
+pub fn cpu_profiler() -> CpuProfiler {
+    #[cfg(feature = "cpuprofiler")]
+    {
+        cpuprofiler::PROFILER.lock().unwrap().start("./out.profile").unwrap();
+    }
+
+    #[cfg(not(feature = "cpuprofiler"))]
+    {
+        eprintln!("cpuprofiler feature is disabled")
```

Comment:
shouldn't this be a `log::info`?

Reply:
I thought about that, but figured that `eprintln` is more reliable. 

The intention is that we only use `cpu_profiler` when actually investigating a particular problem, we shoudn't be adding it everywhere, like we do with `profile` calls: it doesn't support nesting, for example. 

So, if you explicitly put a call to `cpu_profiler` but forgot to enable a feature, it's better to yell loudly